src/driver-hidpp20.c

```diff
@@ -1482,7 +1482,7 @@ hidpp20drv_commit(struct ratbag_device *device)
 	struct ratbag_led *led;
 	struct ratbag_resolution *resolution;
 	int rc;
-	unsigned int dpi_index = 0;
+	unsigned int dpi_index = 0xff;
```

Comment:
AIUI the `dpi_index` is only used in that inner loop, so best to move the declaration to there too so it becomes better contained. And then the `0xff` doesn't look quite as confusing as it does now.